Add profiles to manage docker-compose containers (#334)
gridsuite/deployment: README.md

| security-analysis-server      |        |         |   🗸   |        |  🗸    |        |        |         |
| sensitivity-analysis-server   |        |         |   🗸   |        |  🗸    |        |        |         |
| shortcircuit-server           |        |         |   🗸   |        |  🗸    |        |        |         |
| single-line-diagram-server    |        |         |   🗸   |        |  🗸    |        |        |         |
| study-notification-server     |        |         |   🗸   |        |  🗸    |        |        |         |
| study-server                  |        |         |   🗸   |        |  🗸    |        |        |         |
| timeseries-server             |        |         |   🗸   |        |  🗸    |        |        |         |
| voltage-init-server           |        |         |   🗸   |        |  🗸    |        |        |         |
| case-import-server            |        |         |       |        |       |   🗸    |        |         |

To use a profile, you use simply:
```shell
$ cd docker-compose
$ docker compose --profile suite <cmd>
```

You can also combine multiple profiles:
```shell
$ docker compose --profile study --profile mapping <cmd>
```

> But please note that services/container who belongs to at least one profile can't be accessed if the profile isn't specified.

Reviewer: Why do you say this ? For me it works. 
It's just that you need to be careful for the dependencies according to the doc (`depends_on`).
I missed something ?

Reviewer: Actually `docker compose up` start everything but `down` does not work well if you don't specify the `all` profile...
But `docker compose stop` still works well.

Author: Hum, maybe I wasn't clear in the doc.  
What I want to say is if you do for example `docker compose stop study-server` or `docker compose --profile merging stop study-server` it would not work because study-server is in the `study` and `suite` profiles, so not visible until one of the profiles it is attached (`study` or `suite` ... or `all`) is passed in the cli.

Reviewer: I tried the first one and it works for me.

Author: After some tests, the [auto-starting-profiles](https://docs.docker.com/compose/profiles/#auto-starting-profiles-and-dependency-resolution) documentation seems to be outdated
  * `up` and `down` **need** profiles explicitly
  * `start`/`stop`/`restart` _don't care_ of profiles, they only need the container(s) to be created by a previously `up`
    * and check `depends_on`

➡️ Need to update the readme

> $ docker compose --profile study --profile mapping <cmd>

Reviewer 2: mapping -> dynmap ?

| study-server                  |        |         |   🗸   |      🗸      |        |  🗸    |        |        |         |
| timeseries-server             |        |         |   🗸   |             |        |  🗸    |        |        |         |
| voltage-init-server           |        |         |   🗸   |             |        |  🗸    |        |        |         |
| case-import-server            |        |         |       |             |        |       |   🗸    |        |         |

To use a profile, you use simply:
```shell
$ cd docker-compose
$ docker compose --profile suite <cmd>
```

You can also combine multiple profiles:
```shell
$ docker compose --profile study --profile dynmap <cmd>
```

> [!IMPORTANT]  
> Contrary to what the Docker documentation say:
>   * With the commands `up` & `down`, services/container who belongs to at least one profile can't be accessed if the profile isn't specified.
>     For example `docker compose up study-server` would not work because the profile `study` isn't passed in the CLI.
>     The correct CLI would be `docker compose --profile study up study-server`.

> >   * With the commands `start`, `stop`, `restart`, the `--profile ...` as no effect because theses commands affect the containers already created by a previous `up` command.

Reviewer 2: ```suggestion
>   * With the commands `start`, `stop`, `restart`, the `--profile ...` has no effect because theses commands affect the containers already created by a previous `up` command.
```

| griddyna&#8209;app<br/>dynamic&#8209;mapping&#8209;server | | | | | ✅ | ✅ | ✅ | | | |
| gridmerge&#8209;app<br/>balances&#8209;adjustment&#8209;server<br/>case&#8209;import&#8209;job<br/>case&#8209;validation&#8209;server<br/>cgmes&#8209;assembling&#8209;job<br/>cgmes&#8209;boundary&#8209;import&#8209;job<br/>cgmes&#8209;boundary&#8209;server<br/>merge&#8209;notification&#8209;server<br/>merge&#8209;orchestrator&#8209;server | | ✅ | | | | | ✅ | | | |
| gridstudy&#8209;app<br/>dynamic&#8209;simulation&#8209;server<br/>timeseries&#8209;server | | | ✅ | | | ✅ | ✅ | | | |
| cgmes&#8209;gl&#8209;server<br/>odre&#8209;server<br/>security&#8209;analysis&#8209;server<br/>sensitivity&#8209;analysis&#8209;server<br/>shortcircuit&#8209;server<br/>voltage&#8209;init&#8209;server | | | ✅ | | | | ✅ | | | |
| directory&#8209;notification&#8209;server<br/>directory&#8209;server<br/>explore&#8209;server<br/>geo&#8209;data&#8209;server<br/>gridexplore&#8209;app<br/>network&#8209;map&#8209;server<br/>network&#8209;modification&#8209;server<br/>single&#8209;line&#8209;diagram&#8209;server<br/>study&#8209;notification&#8209;server<br/>study&#8209;server | | | ✅ | ✅ | | ✅ | ✅ | | | |

> | case&#8209;import&#8209;server | | | | | | | | ✅ | | |

Reviewer: IntelliJ propose to reformat this to make it more readable for maintainers :
<details><summary>Details</summary>
<p>

![image](https://github.com/gridsuite/deployment/assets/75525996/f05bfd14-ffc4-4335-a149-2f3618530e72)
![image](https://github.com/gridsuite/deployment/assets/75525996/7d682e27-b772-49f5-9c3c-0fcb55b08a91)

</p>
</details>

Author: yes, I did this because the lines is too long to be readable in the "more readable" form with word warping...

In order to show documents in the case-server index with Kibana, you must first create the index pattern ('Management' page) : case-server*

### Installing / Updating docker-compose to v2
Docker-compose v2 is necessary to have to profiles feature.  

> If possible, prefer to install it with your package manager if your on a Unix system.

Reviewer 2: you're or you are

In order to show documents in the case-server index with Kibana, you must first create the index pattern ('Management' page) : case-server*

### Installing / Updating docker-compose to v2

> Docker-compose v2 is necessary to have to profiles feature.  

Reviewer: ```suggestion
Docker compose v2 is necessary to use the profiles feature.  
```

If possible, prefer to install it with your package manager if you are on a Unix system.

> [!NOTE]  
> You need a client (docker-cli) of ~v19~ v20 at least to have the system of cli-plugins.
> It isn't necessary to update your Docker engine or client else.

> You can install docker-compose with these commands, as instructed in [the doc](https://docs.docker.com/compose/install/linux/#install-the-plugin-manually) and the [migration guide](https://docs.docker.com/compose/migrate/):

Reviewer: ```suggestion
You can install docker compose with these commands, as instructed in [the doc](https://docs.docker.com/compose/install/linux/#install-the-plugin-manually) and the [migration guide](https://docs.docker.com/compose/migrate/):
```

Author: It's the name of [the tool/file](https://github.com/docker/compose).

Reviewer: Well the fille yes, sure, but the tool is called docker compose now, docker-compose is the previous version. We avoid confusion then.

```shell
curl -LR --create-dirs -o $HOME/.local/bin/docker-compose https://github.com/docker/compose/releases/latest/download/docker-compose-linux-x86_64
#wget -x -O $HOME/.local/bin/docker-compose https://github.com/docker/compose/releases/latest/download/docker-compose-linux-x86_64
chmod +x $HOME/.local/bin/docker-compose
mkdir -p $HOME/.docker/cli-plugins
ln -s $HOME/.local/bin/docker-compose $HOME/.docker/cli-plugins/docker-compose
docker compose version
```
You must get this output from docker compose now:

> > Docker Compose version v2.20.3

Reviewer: ```suggestion
> Docker Compose version v2.X.X
```

Author: That is just an example of the output, i don't think it's necessary... no?

Reviewer: For me it's better, otherwise we can think that we need this version specifically
